lib/task_listener.js

```diff
@@ -229,7 +227,18 @@ export default class TaskListener extends EventEmitter {
     }
   }
 
-  addRunningTask(runningState) {
+  async recordCapacity () {
+    this.runtime.stats.record('capacityOverTime', {
+      duration: Date.now() - this.timestamp,
+      availableCapacity: await this.availableCapacity(),
```

Comment:
Yea, we should probably memoize this for at least stats recording.  Most of the time this wouldn't be expensive to call, but because of the way testdroid is currently, this will cause requests to be made to bitbar to determine the available capacity.